rfcs/text/design-tokens.md

```diff
@@ -0,0 +1,154 @@
+- Start Date: (fill me in with today's date, 2021-06-14)
+
+
+- [Summary](#summary)
+- [Basic example](#basic-example)
+- [Motivation](#motivation)
+- [Detailed design](#detailed-design)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+    - [Design tokens via styled-components and CSS variables](#design-tokens-via-styled-components-and-css-variables)
+    - [Tag function for generating token](#tag-function-for-generating-token)
+- [Adoption strategy](#adoption-strategy)
+- [How we teach this](#how-we-teach-this)
+
+# Summary
+
+Implementation of the design tokens with theme support and backwards compatibility.
+
+# Basic example
+
+Design tokens will be kept in `@sage/design-tokens` library. It will be updated every time tokens will change. They will
+be imported from this library and used as the object.
+
+```js
+const Container = styled.div`
+  background: ${props.theme.sagedsColorsBase400};
+  color: ${props.theme.sagedsColorsTextWhite};
+`;
+```
+
+Then, themes would be handled by ThemeProvider.
+
+Themes would be created from existing ones by extending and renaming them, so that current *mint* would be named *sbiz*
+.
+
+```js
+import designTokens from "@sage/design-tokens/js/sbiz/es6-obj";
+import { mergeWithBase } from "../base";
+import configureTheme from "./mint-theme.config";
+
+export default {
+  ...mergeWithBase(configureTheme), // for backwards compability. Eventually should be removed.
+  ...designTokens,
+  name: "sbiz",
+};
+```
+
+# Motivation
+
+Design tokens are basic atomic part of design system. Its purpose is to improve handover process between designers and
+developers. This will result in more consistent ui and will help using colors more
+
+Design tokens are basically key-value pairs named and organized the same way regardless of the platform. This way,
+designers and developers can communicate using key names, and not specific values.
+
+Design tokens can contain values like lengths in different units, colors, font properties, transition definitions and
+others.
+
+Naming of the tokens is consistent across all platforms and provides predictable structure.
+
+To put it in simple words: When developer creates design for component, everything what you, as a developer, care about
+is which token has been used.
+
+# Detailed design
+
+This approach uses themes feature from `styled-components` since it is already implemented and uses javascript
+variables.
+
+At the moment, we have two levels of design tokens:
+
+- **Generic tokens** - kind of palette with all the values and scales.
+- **Theme tokens** - tokens for a theme, which contain references to generic tokens.
+
+So themes can be for example: *S Biz*, *M Biz*, *High Contrast*, *Dark Mode* and so on.
```

Comment:
Where are these defined? What determines the ownership of the themes and how do we handle migration of themes

Reply:
We also need to consider how we support consumers that want to use a non-sage palette. Would they still write a their own theme and pass it to the theme provider or will this change too?

Reply:
Well, using js variables approach, thats what would they do - create own theme using base theme as a template (since it will contain all possible tokens), and then provide it in the Tehem Provider.

Using CSS approach, it would be even easier, since they would create their own themes by defining values for css variables, and then scoping them with classes, or use it globally with `:root{ ... }`